versions/1.0.0.md

```diff
@@ -92,12 +92,12 @@ This object represents one or more changes to be applied to the target document
 
 | Field Name | Type | Description |
 | ---- | :----: | ---- |
-| <a name="action-target"></a>target | `string` | **REQUIRED** A JSONPath query expression referencing the target objects in the target document. |
+| <a name="action-target"></a>target | `string` | **REQUIRED** A JSONPath expression selecting nodes in the target document. |
 | <a name="action-description"></a>description | `string` | A description of the action. [[CommonMark]] syntax MAY be used for rich text representation. |
-| <a name="action-update"></a>update | Any | An object with the properties and values to be merged with the object(s) referenced by the `target`. This property has no impact if `remove` property is `true`. |
+| <a name="action-update"></a>update | Any | If the `target` selects an object node, the value of this field should be an object with the properties and values to merge with the node. If the `target` selects an array, the value of this field should be an entry to append to the array. This field has no impact if the `remove` field of this action object is `true`. |
 | <a name="action-remove"></a>remove | `boolean` | A boolean value that indicates that the target object is to be removed from the the map or array it is contained in. The default value is `false`. |
 
-The result of the `target` JSONPath query expression must be zero or more objects or arrays (not primitive types or `null` values). If you wish to update a primitive value such as a string, the `target` expression should select the _containing_ object in the target document.
+The result of the `target` JSONPath expression must be zero or more objects or arrays (not primitive types or `null` values). If you wish to update or remove a primitive value such as a string, the `target` expression should select the _containing_ object in the target document.
```

Comment:
You replace the array with a new one that does not contain the item to remove. Which means you have to know the full array content.

I definitely want to drop this whole paragraph and allow `target` expressions to select any kind of node, including primitive values, but that would be a significant change since the restriction to objects and arrays has been in since the spec document was created more than three years ago.

Opened
* #57 
to track and discuss this.

Reply:
Then can we just go back to the original wording of that last sentence:
> If you wish to update a primitive value such as a string, the `target` expression should select the _containing_ object in the target document.

Reply:
The second example in [Array Modification Examples](https://github.com/OAI/Overlay-Specification/blob/main/versions/1.0.0.md#array-modification-examples) shows how to remove an object from an array:
~~~yaml
overlay: 1.0.0
info:
  title: Remove a array element
  version: 1.0.0
actions:
  - target: $.paths.*.get.parameters[?@.name == 'dummy']
    remove: true
~~~

Reply:
Got it. You can remove an array element when it is an object using JSON Path filtering.

So do we go back to the original wording for that last sentence? By original wording I mean:
> If you wish to update a primitive value such as a string, the `target` expression should select the _containing_ object in the target document.

We _could_ update this to be even more explicit:
> If you wish to update a primitive value such as a string, the `target` expression should select the _containing_ object in the target document and the action should contain the property with the value to be updated.